src/syntax/ast.ts

```diff
@@ -62,6 +62,18 @@ export interface BeginStatement extends PGNode {
     deferrable?: boolean;
 }
 
+export interface SetTransactionSnapshot extends PGNode {
+    type: 'set transaction snapshot';
+    snapshotId: Name;
+}
+
+export interface SetTransactionMode extends PGNode {
+    type: 'set transaction' | 'set session characteristics as transaction';
+    isolationLevel: 'serializable' | 'repeatable read' | 'read committed' | 'read uncommitted';
+    writeable?: 'read write' | 'read only';
```

Comment:
Should this maybe be a boolean?

Reply:
I agree, if you use `deferable` as boolean, it would make more sense to have a `writeable` as boolean as well.

In fact, given that "writeable" is the default behaviour, it would perhaps be more relevent to have a `readonlyMode: boolean` or something like that ? 

That would allow code like:
```typescript
if (!node.readonlyMode) {
      // do something if writeable
}
```
to behave as expected when the writeability is not explicitely provided